fiftyone/core/odm/database.py

```diff
@@ -786,18 +787,30 @@ def insert_documents(docs, coll, ordered=False, progress=None, num_docs=None):
     return ids
 
 
-def bulk_write(ops, coll, ordered=False):
+def bulk_write(ops, coll, ordered=False, progress=False):
     """Performs a batch of write operations on a collection.
 
     Args:
         ops: a list of pymongo operations
         coll: a pymongo collection
         ordered (False): whether the operations must be performed in order
+        progress (False): whether to render a progress bar (True/False), use
+            the default value ``fiftyone.config.show_progress_bars`` (None), or
+            a progress callback function to invoke instead
     """
+    batcher = fou.get_default_batcher(ops, progress=progress)
+
     try:
-        batch_size = fo.config.bulk_write_batch_size
-        for ops_batch in fou.iter_batches(ops, batch_size):
-            coll.bulk_write(list(ops_batch), ordered=ordered)
+        with batcher:
+            for batch in batcher:
+                batch = list(batch)
+                coll.bulk_write(batch, ordered=ordered)
+                if batcher.manual_backpressure:
+                    # @todo can we infer content size from bulk_write() above?
+                    # @todo do we need a more accurate measure of size here?
```

Comment:
As detailed in a mainline comment, I think this is likely to be good enough. Alternatively we could add size of BSON representation of `_filter` and `_doc` together (I think it is going to send the command to MongoDB as BSON not the string repr which will be slightly different), though not a great idea to use protected attributes.

Reply:
Yeah I think its fine as-is now